packages/lib/modules/lbp/steps/WeightAdjustmentTypeInput.tsx

```diff
@@ -0,0 +1,155 @@
+            <Text color="font.secondary" fontSize="sm">
+              {launchTokenSymbol} 90%
+            </Text>
+            <ArrowRight size={12} />
+            <Text color="font.secondary" fontSize="sm">
+              50%
+            </Text>
+          </HStack>
+        </HStack>
+      ),
+      value: 'linear_90_50',
+    },
+    {
+      label: (
+        <HStack justify="space-between" w="full">
+          <Text>Custom sale</Text>
+        </HStack>
+      ),
+      value: 'custom',
+    },
+  ]
+
+  const weightAdjustment = watch('weightAdjustmentType')
+  const customStartWeight = watch('customStartWeight')
+  const customEndWeight = watch('customEndWeight')
+
+  return (
+    <VStack align="start" w="full">
+      <Text color="font.primary">Dynamic token weight adjustments</Text>
+      <Controller
+        control={control}
+        name="weightAdjustmentType"
+        render={({ field }) => (
+          <SelectInput
+            defaultValue={options[0].value}
+            id="weight-adjustment-type"
+            onChange={newValue => {
+              field.onChange(newValue as GqlChain)
+            }}
+            options={options}
+            value={field.value}
+          />
+        )}
+      />
+
+      {weightAdjustment === 'custom' && (
+        <Stack w="full" mt="md">
+          <WeightSlider
+            title="Starting weights"
+            launchTokenSymbol={launchTokenSymbol}
+            collateralTokenSymbol={collateralTokenSymbol}
+            customWeight={customStartWeight}
+            setValue={setValue}
+          />
+          <WeightSlider
+            title="Ending weights"
+            launchTokenSymbol={launchTokenSymbol}
+            collateralTokenSymbol={collateralTokenSymbol}
+            customWeight={customEndWeight}
+            setValue={setValue}
+          />
+        </Stack>
+      )}
+    </VStack>
+  )
+}
+
+function WeightSlider({
+  title,
+  launchTokenSymbol,
+  collateralTokenSymbol,
+  customWeight,
+  setValue,
+}: {
+  title: string
+  launchTokenSymbol: string
+  collateralTokenSymbol: string
+  customWeight: number
+  setValue: UseFormSetValue<SaleStructureForm>
+}) {
+  return (
+    <VStack align="start" w="full">
+      <Text color="font.primary">{title}</Text>
+      <Box
+        bg="background.level0"
+        border="white"
+        borderRadius="md"
+        shadow="innerBase"
+        sx={{ paddingX: 'md', paddingY: 'lg' }}
+        w="full"
+      >
+        <HStack justifyContent="space-between" spacing="md" w="full">
+          <Text>{`${launchTokenSymbol}: ${customWeight}`}</Text>
+          <Box textAlign="right" w="100px">
+            <Text
+              fontSize="sm"
+              textColor="font.secondary"
+            >{`${collateralTokenSymbol}: ${100 - customWeight}%`}</Text>
+          </Box>
+        </HStack>
+        <SliderWithSteps
+          aria-label="weights-slider"
+          onChange={value => setValue('customStartWeight', value)}
```

Comment:
looks like hard coding `customStartWeight` here is preventing to possibility to edit the ending weights

https://www.loom.com/share/440e35061dee4e639fbb29e3233f111e?sid=c539b21f-e255-4daf-83af-bd35d9b2a5ec

Reply:
You are right, fixed.